File: back/forum/game/merge_lore.py
import json
import os

lore_path = r'c:\laragon\www\foro_local\back\forum\game\lore.json'
lore2_path = r'c:\laragon\www\foro_local\back\forum\game\lore2.txt'

with open(lore_path, 'r', encoding='utf-8') as f:
    lore = json.load(f)

with open(lore2_path, 'r', encoding='utf-8') as f:
    lore2 = json.load(f)

for key in ['eras', 'lore_basal', 'eventos']:
    if key in lore2:
        if key not in lore:
            lore[key] = []
        existing_ids = {item['id'] for item in lore.get(key, []) if 'id' in item}
        for item in lore2[key]:
            if 'id' in item and item['id'] in existing_ids:
                for i, ex_item in enumerate(lore[key]):
                    if ex_item.get('id') == item['id']:
                        lore[key][i] = item
            else:
                lore[key].append(item)

with open(lore_path, 'w', encoding='utf-8') as f:
    json.dump(lore, f, indent=4, ensure_ascii=False)

os.remove(lore2_path)
print('Merge complete and lore2.txt deleted.')

File: back/forum/game/merge_lore.php
<?php
$lore_path = 'c:\\laragon\\www\\foro_local\\back\\forum\\game\\lore.json';
$lore2_path = 'c:\\laragon\\www\\foro_local\\back\\forum\\game\\lore2.txt';

$lore = json_decode(file_get_contents($lore_path), true);
$lore2 = json_decode(file_get_contents($lore2_path), true);

$keys = ['eras', 'lore_basal', 'eventos'];

foreach ($keys as $key) {
    if (isset($lore2[$key])) {
        if (!isset($lore[$key])) {
            $lore[$key] = [];
        }
        
        $existing_ids = [];
        foreach ($lore[$key] as $item) {
            if (isset($item['id'])) {
                $existing_ids[$item['id']] = true;
            }
        }
        
        foreach ($lore2[$key] as $item) {
            if (isset($item['id']) && isset($existing_ids[$item['id']])) {
                foreach ($lore[$key] as $i => $ex_item) {
                    if (isset($ex_item['id']) && $ex_item['id'] == $item['id']) {
                        $lore[$key][$i] = $item;
                    }
                }
            } else {
                $lore[$key][] = $item;
            }
        }
    }
}

file_put_contents($lore_path, json_encode($lore, JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE));
unlink($lore2_path);

echo "Merge complete and lore2.txt deleted.\n";
?>
